src/Engine/ProtoCore/Utils/ArrayUtils.cs

```diff
@@ -132,8 +132,12 @@ public static ClassNode GetGreatestCommonSubclassForArray(StackValue array, Runt
             var dsArray = runtimeCore.Heap.ToHeapObject<DSArray>(array);
             foreach (var sv in dsArray.Values)
             {
+                if(IsEmpty(sv, runtimeCore)) continue;
```

Comment:
We don't want to add a type for an empty list. On the other hand we want to ignore it and inspect the next element of the array.